config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurer.java

```diff
@@ -503,9 +503,14 @@ private AuthenticationEntryPoint getLoginEntryPoint(B http, String providerLogin
 				new OrRequestMatcher(loginPageMatcher, faviconMatcher), defaultEntryPointMatcher);
 		RequestMatcher notXRequestedWith = new NegatedRequestMatcher(
 				new RequestHeaderRequestMatcher("X-Requested-With", "XMLHttpRequest"));
+		RequestMatcher formLoginNotEnabled = (request) -> {
+			DefaultLoginPageGeneratingFilter defaultLoginPageGeneratingFilter = http
```

Comment:
Let's move the variable initialization of `defaultLoginPageGeneratingFilter` outside of the lambda

Reply:
I believe that this only works if this is done inside the lambda. That way, you don't have to worry about the order in which the configurers are registered, as it only matters the outcome and whether Form Login is configured in the end or not.

Reply:
On second thought, it looks like we *should* be able to rely on the `DefaultLoginPageConfigurer` being first, as it is registered by `HttpSecurityConfiguration#httpSecurity()` when subclassing `WebSecurityConfigurerAdapter` in a hello-world style application.